cli/azd/pkg/infra/provisioning/provisioning_plan.go

```diff
@@ -0,0 +1,25 @@
+package provisioning
+
+type ProvisioningPlan struct {
```

Comment:
Let's not use the term `Plan` here since terraform uses that term and our use her differs, I think?  This is really about getting some information from the IaC files themselves (what configuration parameters they have and what outputs they set) vs a plan of resources we to be created.